Lib/lib2to3/pgen2/pgen.py

```diff
@@ -3,17 +3,113 @@
 
 # Pgen imports
 from . import grammar, token, tokenize
+import collections
+
 
 class PgenGrammar(grammar.Grammar):
-    pass
+    def produce_graminit_h(self, writer):
```

Comment:
An alternative way of structuring this would be to have a subclass in Parser/pgen.py that adds this functionality.